### PR TITLE
TSK-1208

### PR DESCRIPTION
**Pull request by krozzus**

### CODE REVIEW

**Comment by mustaphazorgati** [comment]
[TSK-1208]

**Comment by mustaphazorgati** [comment]
[ADPT-70]

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/TaskControllerIntTest.java [hunk 1]
```diff
@@ -421,20 +427,20 @@ void testThrowsExceptionIfInvalidFilterIsUsed() {
...
+    ResponseEntity<TaskanaPagedModel<TaskSummaryRepresentationModel>> response =
         template.exchange(
             restHelper.toUrl(Mapping.URL_TASKS)
                 + "?state=READY,CLAIMED&sort-by=por.value&order=desc&page=15&page-size=5",
             HttpMethod.GET,
             request,
-            ParameterizedTypeReference.forType(TaskSummaryListResource.class));
-    assertThat(response.getBody().getContent()).hasSize(1);
+            TASK_SUMMARY_PAGE_MODEL_TYPE);
+    assertThat(Objects.requireNonNull(response.getBody()).getContent()).hasSize(1);
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/TaskControllerIntTest.java:437]: Please don't do this. 
Add `assertThat(response.getBody()).isNotNull()`
You have done this 16 times. Please replace this everywhere.

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/WorkbasketControllerIntTest.java [hunk 2]
```diff
@@ -21,17 +21,26 @@
...
+import pro.taskana.rest.resource.WorkbasketSummaryRepresentationModel;
 import pro.taskana.workbasket.api.WorkbasketType;
 
 /** Test WorkbasketController. */
 @TaskanaSpringBootTest
 class WorkbasketControllerIntTest {
 
+  private static final ParameterizedTypeReference<
+          TaskanaPagedModel<WorkbasketAccessItemRepresentationModel>>
+      WORKBASKET_ACCESSITEM_PAGE_MODEL_TYPE =
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/WorkbasketControllerIntTest.java:36]: ACCESS_ITEM please :) two words

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/resource/TaskRepresentationModelAssemberTest.java [hunk 3]
```diff
@@ -24,30 +24,32 @@
...
+/** Test for {@link TaskRepresentationModelAssembler}. */
 @TaskanaSpringBootTest
-class TaskResourceAssemberTest {
+class TaskRepresentationModelAssemberTest {
 
   @Autowired TaskService taskService;
   @Autowired WorkbasketService workbasketService;
   @Autowired ClassificationService classificationService;
-  @Autowired TaskResourceAssembler taskResourceAssembler;
+  @Autowired TaskRepresentationModelAssembler taskRepresentationModelAssembler;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/resource/TaskRepresentationModelAssemberTest.java:34]: Can you please use an autowired constructor?

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/resource/TaskSummaryAssemblerTest.java [hunk 4]
```diff
@@ -24,7 +24,7 @@
 class TaskSummaryAssemblerTest {
 
   @Autowired TaskService taskService;
-  @Autowired TaskSummaryResourceAssembler taskSummaryResourceAssembler;
+  @Autowired TaskSummaryRepresentationModelAssembler taskSummaryRepresentationModelAssembler;
   @Autowired WorkbasketService workbasketService;
   @Autowired ClassificationService classificationService;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/resource/TaskSummaryAssemblerTest.java:29]: Can you please use an autowired constructor?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/WorkbasketRepresentationModel.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+@JsonIgnoreProperties("markedForDeletion")
+public class WorkbasketRepresentationModel
+    extends WorkbasketSummaryRepresentationModel {
+
+
+  private String created; // ISO-8601
+  private String modified; // ISO-8601
+
+
+  public WorkbasketRepresentationModel() {
```
[the remark this answers is not in the document]
Comment (krozzus) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/WorkbasketRepresentationModel.java:19]: will be implemented as protected, else the tests and WorkbasketRepresentationModelWithoutLinks cause issues

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/TaskCommentRepresentationModelAssembler.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+  @Override
+  public TaskCommentRepresentationModel toModel(TaskComment taskComment) {
+    TaskCommentRepresentationModel taskCommentRepresentationModel =
+        new TaskCommentRepresentationModel(taskComment);
+    try {
+      taskCommentRepresentationModel.add(
+          linkTo(methodOn(TaskCommentController.class).getTaskComment(taskComment.getId()))
+              .withSelfRel());
+    } catch (TaskCommentNotFoundException
```
[the remark this answers is not in the document]
Comment (krozzus) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/TaskCommentRepresentationModelAssembler.java:48]: As in correspondance with @gitgoodjhe:
in cases in which there are several exceptions that all get treated the same, there is a case to be made for an allencompassing generic exception. however in other cases this is a bad practice and the more specific exception should be caught.

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/WorkbasketRepresentationModel.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
+package pro.taskana.rest.resource;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import pro.taskana.workbasket.api.models.Workbasket;
+
+/**
+ * EntityModel class for {@link Workbasket}.
+ */
+@JsonIgnoreProperties("markedForDeletion")
```
[the remark this answers is not in the document]
Comment (krozzus) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/WorkbasketRepresentationModel.java:10]: this is a bug and has been fixed as part of the latest commit

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/jobs/TaskCleanupJob.java [hunk 8]
```diff
@@ -23,19 +23,21 @@
...
-/** Job to cleanup completed tasks after a period of time. */
+/**
+ * Job to cleanup completed tasks after a period of time.
+ */
 public class TaskCleanupJob extends AbstractTaskanaJob {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(TaskCleanupJob.class);
 
-  private static BaseQuery.SortDirection asc = BaseQuery.SortDirection.ASCENDING;
+  private static final BaseQuery.SortDirection ASCENDING = BaseQuery.SortDirection.ASCENDING;
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/task/internal/jobs/TaskCleanupJob.java:33]: Can you replace this variable with a static import?

File: lib/taskana-core/src/main/java/pro/taskana/workbasket/internal/jobs/WorkbasketCleanupJob.java [hunk 9]
```diff
@@ -76,31 +76,27 @@ public static void initializeSchedule(TaskanaEngine taskanaEngine) {
...
-                  () -> {
-                    try {
-                      return deleteWorkbaskets(workbasketsToBeDeleted);
-                    } catch (Exception e) {
-                      LOGGER.warn("Could not delete workbaskets.", e);
-                      return 0;
-                    }
-                  });
-      return count;
+      return (int) (Integer)
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/workbasket/internal/jobs/WorkbasketCleanupJob.java:90]: Why are you additionally casting from Integer to int? Autounboxing should do this automatically

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/TaskCommentRepresentationModelAssembler.java [hunk 10]
```diff
@@ -81,6 +76,6 @@ public TaskComment toEntityModel(TaskCommentRepresentationModel taskCommentRepre
                .collect(
                    Collectors.collectingAndThen(
                        Collectors.toList(),
-                       list -> new TaskanaPagedModel<>(TASK_COMMENTS, list, pageMetadata)));
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/TaskCommentRepresentationModelAssembler.java]: Personally I'd keep the enum name with an underscore

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/TaskanaPagedModelKeys.java [hunk 11]
```diff
@@ -11,7 +11,7 @@
   CLASSIFICATIONS("classifications"),
   DISTRIBUTIONTARGETS("distributionTargets"),
   TASKS("tasks"),
-  TASK_COMMENTS("task_comments"),
+  TASKCOMMENTS("taskComments"),
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/TaskanaPagedModelKeys.java:14]: as mentionned above: I prefer TASK_COMMENTS

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/TaskanaPagedModelKeys.java [hunk 12]
```diff
@@ -11,7 +11,7 @@
   CLASSIFICATIONS("classifications"),
   DISTRIBUTIONTARGETS("distributionTargets"),
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/resource/TaskanaPagedModelKeys.java:12]: DISTRIBUTION_TARGETS would be a better name. (personal preference)

File: rest/taskana-rest-spring/src/test/java/pro/taskana/doc/api/ClassificationControllerRestDocumentation.java [hunk 13]
```diff
@@ -248,10 +250,11 @@ void setUp() {
 
   @Test
   void getAllClassificationsDocTest() throws Exception {
+    classificationFieldDescriptionsMap.put("_links.self.href", "link to self");
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/doc/api/ClassificationControllerRestDocumentation.java:253]: why did you add this?

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/WorkbasketAccessItemControllerIntTest.java [hunk 14]
```diff
@@ -47,20 +47,23 @@ void testGetAllWorkbasketAccessItems() {
...
     ResponseEntity<TaskanaPagedModel<WorkbasketAccessItemRepresentationModel>> response =
         template.exchange(
             restHelper.toUrl(Mapping.URL_WORKBASKETACCESSITEMS) + parameters,
             HttpMethod.GET,
             restHelper.defaultRequest(),
-            WORKBASKET_ACCESSITEM_PAGE_MODEL_TYPE);
+            WORKBASKET_ACCESS_ITEM_PAGE_MODEL_TYPE);
+    assertThat(response.getBody()).isNotNull();
     assertThat(response.getBody().getLink(IanaLinkRelations.SELF)).isNotNull();
+    System.out.println(response);
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/WorkbasketAccessItemControllerIntTest.java:66]: remove this please

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/WorkbasketAccessItemControllerIntTest.java [hunk 15]
```diff
@@ -90,17 +91,19 @@ void testThrowsExceptionIfInvalidFilterIsUsed() {
...
             HttpMethod.GET,
             restHelper.defaultRequest(),
-            WORKBASKET_ACCESSITEM_PAGE_MODEL_TYPE);
+            WORKBASKET_ACCESS_ITEM_PAGE_MODEL_TYPE);
+    assertThat(response.getBody()).isNotNull();
     assertThat(response.getBody().getContent()).hasSize(1);
     assertThat(response.getBody().getContent().iterator().next().getAccessId())
         .isEqualTo("user_1_1");
     assertThat(response.getBody().getLink(IanaLinkRelations.SELF)).isNotNull();
+    System.out.println(response);
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/WorkbasketAccessItemControllerIntTest.java:106]: remove this please

File: web/package-lock.json [hunk 16]
```diff
@@ -7017,11 +7017,6 @@
       "resolved": "https://registry.npmjs.org/graceful-fs/-/graceful-fs-4.2.3.tgz",
       "integrity": "sha512-a30VEBm4PEdx1dRB7MFK7BejejvCvBronbLjht+sHuGYj8PHs7M/5Z+rt5lw551vZ7yfTCj4Vuyy3mSJytDWRQ=="
     },
-    "hammerjs": {
-      "version": "2.0.8",
-      "resolved": "https://registry.npmjs.org/hammerjs/-/hammerjs-2.0.8.tgz",
-      "integrity": "sha1-BO93hiz/K7edMPdpIJWTAiK/YPE="
-    },
```
Comment (mustaphazorgati) [inline web/package-lock.json:7024]: why did you remove this?

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/jobs/TaskCleanupJob.java [hunk 17]
```diff
@@ -30,7 +30,7 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(TaskCleanupJob.class);
 
-  private static final BaseQuery.SortDirection ASCENDING = BaseQuery.SortDirection.ASCENDING;
+  private static final SortDirection ASCENDING = SortDirection.ASCENDING;
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/task/internal/jobs/TaskCleanupJob.java:33]: That's not what I meant.. But it's fine.. no need for further refactoring
